js/components/dataset/resource/file-form.vue

```diff
@@ -102,11 +102,25 @@ export default {
     },
     computed: {
         upload_endpoint: function() {
-            let endpoint = API.datasets.operations.upload_dataset_resource;
+            let operations = API.datasets.operations;
+            let params = {};
+            let endpoint = 'upload_';
+            if (typeof this.dataset !== 'undefined') {
+                params = {dataset: this.dataset.id};
+            }
             if (this.community) {
-                endpoint = API.datasets.operations.upload_community_resource;
+                endpoint += 'community_';
+                params.community = this.resource.id;
+            } else {
+                endpoint += 'dataset_';
             }
-            return endpoint.urlify({dataset: this.dataset.id});
+            if (this.resource.id) {
+                endpoint += 'resource';
+                params.rid = this.resource.id;
+            } else {
+                endpoint += 'resources';
+            }
```

Comment:
How many endpoints/cases are there ?
Is'nt it possible to concat everything in a single if/else ?